### ISSUE TITLE
Timeout with large KVM virtual machines not being imported

### ISSUE THREAD

**Issue by pcfriek1987** [apache/cloudstack#8819]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
UI / Migration
~~~

##### CLOUDSTACK VERSION

~~~
4.19
~~~

##### CONFIGURATION

N/A
##### OS / ENVIRONMENT

N/A

##### SUMMARY

When migrating a virtual machine from a remote KVM which has a large disk a timeout occurs, on the remote host you can see the qemu convert option running which I think takes too long and then SSH times out. 

##### STEPS TO REPRODUCE

~~~
Have  remote KVM host and try to import a large KVM vm (here it stopped at 47GB already) but might depend on the speed of the storage too.

~~~

##### EXPECTED RESULTS

~~~
That the virtual machine imported succesfully
~~~

##### ACTUAL RESULTS

~~~
On the GUI the following is displayed:

"Failed to import volumes while importing vm: test vm. Error while copying volume".

On the agent log of the cloudstack server it's trying to copy it too you can see the following:
2024-03-21 14:05:28,673 ERROR [utils.ssh.SshHelper] (agentRequest-Handler-5:null) (logid:55e6fedb) Timed out in waiting for SSH execution exit status
2024-03-21 14:05:28,674 ERROR [resource.wrapper.LibvirtCopyRemoteVolumeCommandWrapper] (agentRequest-Handler-5:null) (logid:55e6fedb) Error while copying file from remote host: com.cloud.utils.ssh.SshException: Timed out in waiting for SSH execution exit status

In the journal log on the remote host you can see it connected on SSH:
Mar 21 14:03:28 sshd[2790590]: Accepted password for root from IP port 44740 ssh2
Mar 21 14:03:28 sshd[2790590]: pam_unix(sshd:session): session opened for user root(uid=0) by (uid=0)
Mar 21 14:03:28 systemd-logind[897]: New session 601 of user root.
Mar 21 14:03:28 systemd[1]: Started session-601.scope - Session 601 of User root.
Mar 21 14:03:28 sshd[2790590]: pam_env(sshd:session): deprecated reading of user environment enabled
Mar 21 14:05:28 sshd[2790590]: Received disconnect from IP port 44740:11: Closed due to user request.

Already tried to try and have a longer SSH session by setting

ClientAliveInterval 600
ClientAliveCountMax 6

in the SSHD file but that didn't change much, but maybe I need to change a different setting?
~~~

**Comment by kiranchavala** [comment]
CloudStack uses trilead library for ssh connections. It has a connection timeout, but there is no option to specify timeout for the transfer.

We can consider for this as improvement request

**Comment by sureshanaparti** [comment]
@pcfriek1987 The default result wait timeout was 120 secs (2 mins), so the connection was dropped. The same can be noticed from your journal logs.

```
Mar 21 14:03:28 sshd[2790590]: Accepted password for root from IP port 44740 ssh2
Mar 21 14:03:28 sshd[2790590]: pam_unix(sshd:session): session opened for user root(uid=0) by (uid=0)
Mar 21 14:03:28 systemd-logind[897]: New session 601 of user root.
Mar 21 14:03:28 systemd[1]: Started session-601.scope - Session 601 of User root.
Mar 21 14:03:28 sshd[2790590]: pam_env(sshd:session): deprecated reading of user environment enabled
Mar 21 14:05:28 sshd[2790590]: Received disconnect from IP port 44740:11: Closed due to user request.
```

The copy timeout can be configured through the new config introduced in this PR: #9213

**Comment by pcfriek1987** [comment]
Mine are migrated now but it's a good thing to have for new migrations in the future.

**Comment by sureshanaparti** [comment]
Fixed in #9213
